packages/flutter_tools/lib/src/build_system/targets/ios.dart

```diff
@@ -553,6 +553,50 @@ class ThinIosApplicationFrameworks extends Target {
   @override
   List<Source> get outputs => const <Source>[];
 
+  @override
+  bool canSkip(Environment environment) {
+    final String requestedArchs = environment.defines[kIosArchs];
+    if (requestedArchs == null) {
+      return false;
+    }
+    final Set<String> requestedArchList = requestedArchs.split(' ').toSet();
+    return _setEquals(_existingArchs(environment), requestedArchList);
+  }
+
+  /// Architectures in the original Flutter.framework.
+  Set<String> _existingArchs(Environment environment) {
+    if (_existingArchsByEnvironment[environment] == null) {
+      // Example outputs ("Architectures in the fat file" output has a trailing space):
+      // Architectures in the fat file: /project/build/ios/Debug-iphoneos/Flutter.framework/Flutter are: armv7 arm64
+      // Non-fat file: /project/build/ios/Debug-iphoneos/Flutter.framework/Flutter is architecture: arm64
+      final String lipoOutput = _lipoOutput(environment);
+      final List<String> parts = lipoOutput?.split(': ')?.toList();
+      if (parts == null || parts.length < 2) {
+        return null;
+      }
+      _existingArchsByEnvironment[environment] = parts.last.trim().split(' ').toSet();
+    }
+    return _existingArchsByEnvironment[environment];
+  }
+  final Map<Environment, Set<String>> _existingArchsByEnvironment = <Environment, Set<String>>{};
+
+  String _lipoOutput(Environment environment) {
```

Comment:
Why is the cache/lack of const necessary?

Reply:
So `lipo -info` isn't called twice, once to see if it can be skipped, then a second time when `build` is called.

However that call is extremely fast (on my machine `0.00s user 0.00s system 73% cpu 0.012 total`) so I can be convinced the extra complexity/lack of const isn't worth it.

Reply:
maybe it is worth it, but then you'd need to be sure that  any targets that declare this as a dependency use the same instance of this object - otherwise you will repeat the work anyway

Reply:
generally I would say ensure correctness first and then hit performance later. Since the rest of the target instances are used in a const manner I would say its safer to keep this class more like them